packages/sdks-tests/src/snippet-tests/advanced-child.spec.ts

```diff
@@ -25,25 +25,30 @@ test.describe('Advanced child sub components', () => {
   });
 
   test('Display content for the clicked tab and hide the other', async ({ page, packageName }) => {
-    test.skip(!['react', 'angular', 'angular-ssr'].includes(packageName));
+    test.skip(!['react', 'angular', 'angular-ssr', 'gen1-remix'].includes(packageName));
 
     await page.goto('/advanced-child');
 
     await page.waitForSelector('button:has-text("Tab 1")');
     await page.waitForSelector('button:has-text("Tab 2")');
 
-    await verifyTabContent(
-      page,
-      'Tab 1',
-      'component.options.tabList.0.blocks',
-      'component.options.tabList.1.blocks'
-    );
+    await page.click('button:has-text("Tab 1")');
 
-    await verifyTabContent(
-      page,
-      'Tab 2',
-      'component.options.tabList.1.blocks',
-      'component.options.tabList.0.blocks'
-    );
+    const tab1Content = await page.waitForSelector('div:has-text("Tab 1 Content")', {
+      state: 'visible',
+    });
+    expect(await tab1Content.isVisible()).toBe(true);
+
+    const tab2ContentHidden = await page.$('div:has-text("Tab 2 content")');
+    expect((await tab2ContentHidden?.isVisible()) ?? false).toBe(false);
```

Comment:
`expect((await element?.isVisible()) ?? false).toBe(false)` is confusing and hard to follow. Also, using `page.$` is not recommended, we should rely on `page.locator` whenever possible.

based on https://github.com/microsoft/playwright/issues/18815#issuecomment-1315751818

you should be able to do:

```suggestion
    expect(await page.locator('div').filter({ hasText: 'Tab 2 content' }).isVisible()).toBeFalsy();
```

Reply:
@samijaber I have made it much simpler in the test case by calling a helper function. Let me know if that is okay.